torch/distributed/_tensor/random.py

```diff
@@ -32,11 +32,12 @@ def set_rng_state(new_state: Tensor, device_mesh: DeviceMesh) -> None:
 
     if device_mesh.get_coordinate() is not None:
         # the current rank is in mesh
-        if device_mesh.device_type == "cuda":
-            torch.cuda.set_rng_state(new_state)
+
+        if device_mesh._device_handle:
```

Comment:
similarly in here, we can create `device_handle` again so that we don't need to save it to device_mesh itself.